<launch>
  <!-- CAMERA NODE -->
  <node pkg="usb_cam" type="usb_cam_node" name="barc_cam" output="screen">
    <param name="pixel_format" value="yuyv"/>
    <param name="sharpness" value="255"/>
    <param name="saturation" value="255"/>
    <param name="framerate" value="30"/>    
    <param name="contrast" value="5"/>    
  </node>

  <!-- IMAGE PROCESSING NODE -->
  <node pkg="barc" type="image_processing.py" name="image_processing" output="screen">
    <param name="display_image" value="True" type="boolean"/>
    <param name="calibrate_transform" value="False" type="boolean"/>
    <param name="upperLeftX" value="275" type="int"/>
    <param name="upperLeftY" value="198" type="int"/>
    <param name="upperRightX" value="383" type="int"/>
    <param name="upperRightY" value="196" type="int"/>
    <param name="lowerLeftX" value="45" type="int"/>
    <param name="lowerLeftY" value="393" type="int"/>
    <param name="lowerRightX" value="619" type="int"/>
    <param name="lowerRightY" value="392" type="int"/>
  </node>
  
  <!-- IMU NODE -->
  <node pkg="barc" type="imu_data_acquisition.py" name="imu_node" >
    <param name="port" value="/dev/ttyACM0" />
  </node>

  <!-- ARDUINO NODE -->
  <!--  * encoders and ultrasound sensors -->
  <node pkg="rosserial_python" type="serial_node.py" name="arduino_node" >
    <param name="port" value="/dev/ttyUSB0" />
  </node>

  <!-- STEERING CONTROLLER NODE -->
  <node pkg="barc" type="lane_keeping_steering_control.py" name="steering_controller" output="screen">
    <!-- PROPERTIES -->
    <param name="v_x_pwm" type="int" value="90" />
    <param name="Kp" value="0.1" />
    <param name="Kd" value="0.0" />
    <param name="Ki" value="0.0" />
  </node>

  <?ignore
  <node pkg="rosbag" type="record" name="rosbag_record"
        args=" --all" />
  ?>
</launch>
